src/utils.js

```diff
@@ -13,20 +13,23 @@ export const getComponentVariant = (theme, componentName, variant) => {
 
 export const getComponentStyle = componentName => themeGet(`components.${componentName}.style`);
 
-export const getComponentClassName = ({ className, theme: { classPrefix }, variant }, name) =>
-  `${className || ''} ${classPrefix}-${name} ${variant ? `${classPrefix}-${name}-${variant}` : ''}`.trim();
+const getComponentClassName = ({ baseClassName, theme: { classPrefix }, variant }, name) =>
+  `${baseClassName || ''} ${classPrefix}-${name} ${variant ? `${classPrefix}-${name}-${variant}` : ''}`.trim();
 
 export const createComponent = ({ name, tag = 'div', as, style, props: baseProps = () => ({}) }) => {
   const component = as ? styled(as) : styled[tag];
 
   return component.attrs(props => {
-    const resolvedProps = {
-      ...baseProps(props),
-      ...props,
+    const resolvedProps = { ...baseProps(props), ...props };
+
+    const finalProps = {
+      ...resolvedProps,
+      baseClassName: get(resolvedProps, 'baseClassName', props.className),
```

Comment:
Nice! This looks good. Sorry to be annoying but one more perf suggestion to only spread once:
```js
const baseProps = getBaseProps(props);
const finalProps = {
   ...baseProps,
   ...props,
    baseClassName: get(baseProps, 'baseClassName', props.className),
}
```

Reply:
no problem!